### PR TITLE
Migration: first version of migration script

### PR DESCRIPTION
**Pull request by josojo**
Hey,

this is the version for the migration script. This is not intended to be the final migration script. 

Following points can be discussed:
Best practices on: Importing other smart contracts + their migration scripts/ 
Is there a better solution than DevDeppendencies.sol/ Update on Alan’s pr.

Gnosis-utils vs global standard of zeppelin Which ERC20/EtherToken etc.

I put them on the agenda for the smartcontract weekly.

### CODE REVIEW

File: contracts/DevDependencies.sol [hunk 1]
```diff
@@ -0,0 +1,15 @@
+pragma solidity ^0.4.21;
+
+// NOTE:
+//  This file porpouse is just to make sure truffle compiles all of depending
```
Comment (fleupold) [inline contracts/DevDependencies.sol]: nit: purpose

File: contracts/Plasma.sol [hunk 2]
```diff
@@ -142,7 +142,7 @@ contract Plasma {
         currentFeeExit = 1;
         // Support only ETH on deployment; other tokens need
         // to be added explicitly.
-        listedTokens[0] = WETH;
+        listedTokens.push(WETH);
         exitsQueues[0] = address(new PriorityQueue());
```
Comment (fleupold) [inline contracts/Plasma.sol:146]: Why don't we push exitQueues as well? Also, this changes seems unrelated to the purpose of this PR, so maybe just revert?

File: migrations/2_DEP_migration.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
+
+/* eslint no-undef: "error" */
+//const deployMath = require("@gnosis.pm/util-contracts/src/migrations/2_deploy_math")
```
Comment (fleupold) [inline migrations/2_DEP_migration.js]: remove?

File: migrations/2_DEP_migration.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+module.exports = function (deployer, network, accounts) {
+  	if (network === "development") {
+	    const deployParams = {
+	      artifacts,
+	      deployer,
+	      network,
+	      accounts
+	    }
+	     deployer
+	    //  .then(() => deployMath(deployParams))
```
Comment (fleupold) [inline migrations/2_DEP_migration.js]: ditto

File: migrations/2_DEP_migration.js [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+	      .then(() => deployWeth(deployParams))
+	  	} else {
```
Comment (fleupold) [inline migrations/2_DEP_migration.js]: nit: indentation

File: contracts/DevDependencies.sol [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+//  contracts when we are in development.
+// 
+//  For other environments, we just use the compiled contracts from the NPM 
+//  package
+ 
+ 
+// TODO: Use the same getter pattern also for dependencies
+import "@gnosis.pm/util-contracts/contracts/StandardToken.sol";
```
Comment (fleupold) [inline contracts/DevDependencies.sol]: I don't fully understand why we need this? Wouldn't these get compiled automatically if they are used somewhere in our code?

File: truffle.js [hunk 7]
```diff
@@ -1,18 +1,24 @@
...
- * mainnet: {
- *     provider: function() { 
- *       return new HDWalletProvider(mnemonic, 'https://mainnet.infura.io/<infura-key>') 
- *     },
- *     network_id: '1',
- *     gas: 4500000,
- *     gasPrice: 10000000000,
- *   },
- */
+const truffleConfig = require('@gnosis.pm/util-contracts/src/util/truffleConfig')
```
Comment (fleupold) [inline truffle.js:1]: Is it good practice to include this file? Shouldn't developers set up their own deployment config?

**Review by fleupold — APPROVED** [review]
Mostly questions and small style comments inline. Can we hold off merging until after the smart contract weekly today (when the topics are discussed)?

File: contracts/DevDependencies.sol [hunk 8]
```diff
@@ -0,0 +1,15 @@
+pragma solidity ^0.4.21;
```
Comment (bh2smith) [inline contracts/DevDependencies.sol:1]: Would it be better to use the most recent version of Solidity for all new contracts?

File: contracts/Plasma.sol (hunk 2 above)
Comment (bh2smith) [reply contracts/Plasma.sol:146]: @fleupold  I think that these are two different data types (one is a list and the other is a mapping).

Comment (josojo) [reply contracts/Plasma.sol:146]: yes, one is an array and the other is a mapping. Needed to introduce it here, since otherwise the constructor throws an error

File: migrations/2_DEP_migration.js (hunk 3 above)
Comment (josojo) [reply migrations/2_DEP_migration.js]: It depends on the outcome of the smartcontract call. I am not sure which math file, we want to use

File: contracts/DevDependencies.sol (hunk 6 above)
Comment (josojo) [reply contracts/DevDependencies.sol]: Yes, this one I can take out, the EtherToken will be needed

File: truffle.js (hunk 7 above)
Comment (josojo) [reply truffle.js:1]: I think it is nice to share the config over all projects. I don't see a reason, why we should change any config besides different global gas variables, etc...

File: contracts/DevDependencies.sol (hunk 8 above)
Comment (josojo) [reply contracts/DevDependencies.sol:1]: as long as the ^ is put in there, it shouldn't make a difference

File: contracts/DevDependencies.sol (hunk 6 above)
Comment (fleupold) [reply contracts/DevDependencies.sol]: Why is EtherToken needed? Wouldn't the first time we are using it in code lead to it being compiled?

**Review by bh2smith — COMMENTED** [review]
Seems fine to me for now!
